recipes/google-cloud-cpp-compute/meta.yaml

```diff
@@ -0,0 +1,128 @@
+{% set version = "2.17.0" %}
+
+package:
+  name: google-cloud-cpp-compute
+  version: {{ version }}
+
+source:
+  url: https://github.com/googleapis/google-cloud-cpp/archive/v{{ version }}.tar.gz
+  sha256: 8cb87ec2947e691a7f8631877e78453bcfda51b3d3cd4940794a376741888d37
+  patches:
+    # This is an upstream fix to allow sharding google/cloud/oauth2
+    - patches/0001-Backport-12911-from-upstream.patch
+
+build:
+  number: 0
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+    - libgrpc
+    - libprotobuf
+    - libgoogle-cloud-devel =={{ version }}
```

Comment:
Why is this in build?

Reply:
Because I still don't have a firm grasp of what goes in `build` vs. `host` vs. `run`.  Removed.

Reply:
Without it, the local build fails with:

```
CMake Error at CMakeLists.txt:254 (find_package):
  Could not find a package configuration file provided by
  "google_cloud_cpp_common" with any of the following names:

    google_cloud_cpp_commonConfig.cmake
    google_cloud_cpp_common-config.cmake

  Add the installation prefix of "google_cloud_cpp_common" to
  CMAKE_PREFIX_PATH or set "google_cloud_cpp_common_DIR" to a directory
  containing one of the above files.  If "google_cloud_cpp_common" provides a
  separate development package or SDK, be sure it has been installed.
```

`google-cloud-cpp-compute` depends on some of the libraries created in the `google-cloud-cpp-core-feedstock`.

I still don't have a firm grasp of what goes in `build` vs. `host` vs. `run`. So if it really belongs elsewhere I am happy to change it.  Or if you can suggest some documentation / tutorial that explains how each ought to be used, and would be thankful for that too.

Reply:
> I still don't have a firm grasp of what goes in `build` vs. `host` vs. `run`. So if it really belongs elsewhere I am happy to change it. Or if you can suggest some documentation / tutorial that explains how each ought to be used, and would be thankful for that too.

It's a thing that's hard to grasp indeed. I think the cross-compilation-case is actually the most instructive. Imagine a x64 agent, but we want to compile a library for aarch64. That means the stuff we need in the build environment (compiler, linker, etc. - everything that's actually executed during the build) needs to match the x64 architecture.

On the other hand, the host environment needs to match the target architecture (here aarch64), and any libraries we're compiling against must be in the host-environment, with the right ABI. Compiling against a static or header-only library _only_ requires a host-dependence (everything necessary gets consumed during compilation), whereas compiling against a shared library generates a runtime requirement (a "run-export" from `host:` to `run:`), which ensures that the actual DSO that's necessary to provide the symbols at runtime will be present. That's why we need `libgoogle-cloud-X-devel` in host, but only run-export `libgoogle-cloud-X`.

While there are some exceptions (e.g., we might need `libprotobuf` in the build environment for cross-compilation, because we need to execute `protoc`, which we can only do with the CPU arch of the agent), it's generally a sign of something going wrong to include "libraries I need to compile against" in the `build` environment. What's likely happening is that something somewhere is using a relative lookup (or some other heuristic, e.g. relative to the compilers) that expects the libraries in `$BUILD_PREFIX`, but we really should point that logic to `$PREFIX` (==host). In CMake that's usually something like `CMAKE_PREFIX_PATH=$PREFIX`.

Now the above focussed on the cross-compilation part for illustration purposes, but it holds as well for the native compilation case. The build environment is where the compilers and other build tools live, which are unrelated (including any transitive dependencies they might drag along) from the libraries we're compiling against.

Reply:
Thanks, that was useful.   Let me know if it all looks good now.  I will change the other PRs once we have an agreed structure for them.